src/test/java/com/atlassian/db/replica/api/ThrottledCacheTest.java

```diff
@@ -0,0 +1,75 @@
+package com.atlassian.db.replica.api;
+
+import org.junit.Test;
+
+import java.util.ConcurrentModificationException;
+import java.util.Random;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatNoException;
+
+public class ThrottledCacheTest {
+    private final ExecutorService executor = Executors.newFixedThreadPool(10);
+
+    @Test
+    public void staleWhileInvalidating() throws InterruptedException {
+        ThrottledCache<Long> cache = new ThrottledCache<>();
+
+        cache.get(() -> 1L);
+        asyncPutWithSlowSupplier(cache, anyValue());
+
+        assertThat(cache.get(this::anyValue)).hasValue(1L);
+    }
+
+
+    @Test
+    public void serveLatestValue() {
+        ThrottledCache<Long> cache = new ThrottledCache<>();
+        cache.get(this::anyValue);
+        cache.get(this::anyValue);
+
+        assertThat(cache.get(() -> 4L)).hasValue(4L);
+    }
+
+    @Test
+    public void noConcurrentInvalidation() throws InterruptedException {
+        ThrottledCache<Long> cache = new ThrottledCache<>();
+
+        asyncPutWithSlowSupplier(cache, 2L);
```

Comment:
assertion For `2L` ?

Reply:
I guess it doesn't matter, whole point here is to prove that no exception will be thrown.

Reply:
TBH it should be anyValue - not important for test case.